jcl/src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1525,6 +1551,11 @@ void uncaughtException(Throwable e) {
  * @see J9VMInternals#threadCleanup()
  */
 void cleanup() {
+/*[IF Java14]*/
+	/* Refresh deadInterrupt value so it is accurate when vm thread is cleaned up. */	
+	deadInterrupt = interrupted();
```

Comment:
I like this approach.  Can you confirm that the thread is "isAlive" at this point?  Otherwise the interrupted() call will return the previous setting of `deadInterrupt`.

Reply:
The thread will still be considered alive at this point. isAlive will start to return false when threadRef is cleared: https://github.com/eclipse/openj9/blob/5a2386c150d3277d326c05224b206ef041ffbc73/jcl/src/java.base/share/classes/java/lang/Thread.java#L1545